### PR TITLE
chore(apigatewayv2): rename 'domainName' to 'name' in the DomainName construct

### PR DESCRIPTION
**Pull request by nija-at**
C# does not allow a property to have the same name as its class.
Jsii works around this by renaming the construct to `_DomainName` which
is ugly.

Instead, change the property name from `domainName` to `name` and avoid
this conflict entirely.

Further, re-categorize these class of defects as an error instead of a
warning.

BREAKING CHANGE: `domainName` property under `DomainName` has been
renamed to `name`.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-apigatewayv2/lib/common/domain-name.ts [hunk 1]
```diff
@@ -70,32 +66,19 @@ export interface DomainNameProps {
...
-   * The custom domain name for your API in Amazon API Gateway.
-   *
-   * @attribute
-   */
-  public readonly domainName: string;
-
-  /**
-   * The domain name associated with the regional endpoint for this custom domain name.
-   */
+  public readonly name: string;
```
Comment (NetaNir) [inline packages/@aws-cdk/aws-apigatewayv2/lib/common/domain-name.ts:80]: why did you remove the docs? Specifically, the `@attribute?`

Comment (nija-at) [reply packages/@aws-cdk/aws-apigatewayv2/lib/common/domain-name.ts:80]: They are automatically inherited from the interface `IDomainName`.

Comment (nija-at) [reply packages/@aws-cdk/aws-apigatewayv2/lib/common/domain-name.ts:80]: This is also why the linter does not complain.

**Review by NetaNir — APPROVED** [review]
